emukit/core/loop/outer_loop.py

```diff
@@ -51,8 +51,8 @@ def __init__(self, candidate_point_calculator: CandidatePointCalculator,
         if self.loop_state is None:
             self.loop_state = LoopState([])
 
-        self.loop_start_event = EventHandler()
-        self.iteration_end_event = EventHandler()
+        self.loop_start_event = []
```

Comment:
I don't think this is necessary. The `EventHandler` already acts as a list. You can do `self.loop_start_event.append(new_event_fcn)` and doing `self.iteration_end_event(self, self.loop_state)` with call all functions attached to the event. Let me know if I'm missing something though

Reply:
That is correct, and is the primary reason EventHandler was introduced: to handle multiple event handlers

Reply:
ah I see, that makes sense